src/garage/tf/policies/categorical_gru_policy2.py

```diff
@@ -0,0 +1,301 @@
+
+        """
+        with tf.compat.v1.variable_scope(self.name) as vs:
+            self._variable_scope = vs
+            step_input_var = tf.compat.v1.placeholder(shape=(None,
+                                                             self._input_dim),
+                                                      name='step_input',
+                                                      dtype=tf.float32)
+            step_hidden_var = tf.compat.v1.placeholder(
+                shape=(None, self._hidden_dim),
+                name='step_hidden_input',
+                dtype=tf.float32)
+            self.model.build(state_input,
+                             step_input_var,
+                             step_hidden_var,
+                             name=name)
+
+        self._f_step_prob = tf.compat.v1.get_default_session().make_callable(
+            [
+                self.model.networks['default'].step_output,
+                self.model.networks['default'].step_hidden
+            ],
+            feed_list=[step_input_var, step_hidden_var])
+
+    @property
+    def vectorized(self):
+        """Vectorized or not.
+
+        Returns:
+            Bool: True if primitive supports vectorized operations.
+
+        """
+        return True
+
+    def reset(self, dones=None):
+        """Reset the policy.
+
+        Note:
+            If `dones` is None, it will be by default np.array([True]), which
+            implies the policy will not be "vectorized", i.e. number of
+            paralle environments for training data sampling = 1.
+
+        Args:
+            dones (numpy.ndarray): Bool that indicates terminal state(s).
+
+        """
+        if dones is None:
+            dones = [True]
+        dones = np.asarray(dones)
+        if self._prev_actions is None or len(dones) != len(self._prev_actions):
+            self._prev_actions = np.zeros(
+                (len(dones), self.action_space.flat_dim))
+            self._prev_hiddens = np.zeros((len(dones), self._hidden_dim))
+
+        self._prev_actions[dones] = 0.
+        self._prev_hiddens[dones] = self.model.networks[
+            'default'].init_hidden.eval()
+
+    def get_action(self, observation):
+        """Return a single action.
+
+        Args:
+            observation (numpy.ndarray): Observations.
+
+        Returns:
+            int: Action given input observation.
+            dict(numpy.ndarray): Distribution parameters.
+
+        """
+        actions, agent_infos = self.get_actions([observation])
+        return actions[0], {k: v[0] for k, v in agent_infos.items()}
+
+    def get_actions(self, observations):
+        """Return multiple actions.
+
+        Args:
+            observations (numpy.ndarray): Observations.
+
+        Returns:
+            list[int]: Actions given input observations.
+            dict(numpy.ndarray): Distribution parameters.
+
+        """
+        if self._state_include_action:
+            assert self._prev_actions is not None
```

Comment:
I think this is dangerous -- how do we know that the observations have a stable ordering? I think instead this interface should accept `prev_agent_info`. then the agent can pass whatever it wants to itself between time steps.

Reply:
@krzentner WDYT?

Reply:
this applies to all the recurrent policies, but I will only comment on it here.

Reply:
we already have an issue for this #1012 . I think this should be fixed in another PR.